tests/python/pants_test/projects/test_testprojects_integration.py

```diff
@@ -102,47 +101,98 @@ def targets(self):
     ]
 
     simply_skip = [
-      # Already tested at pants_test.backend.jvm.targets.test_jar_dependency_integration.JarDependencyIntegrationTest
+      # Already tested at
+      # pants_test.backend.jvm.targets.test_jar_dependency_integration.JarDependencyIntegrationTest
       'testprojects/3rdparty/org/pantsbuild/testprojects:testprojects',
       # Already tested in 'PantsRequirementIntegrationTest' and 'SetupPyIntegrationTest'.
       'testprojects/pants-plugins/*',
       'testprojects/src/python/python_distribution/ctypes:ctypes_test',
       'testprojects/src/python/python_distribution/ctypes_with_third_party:ctypes_test',
       'testprojects/src/python/python_distribution/ctypes_with_extra_compiler_flags:bin',
-      # Requires non-standard settings, and already tested by `ProtobufIntegrationTest.test_import_from_buildroot`
+      # Requires non-standard settings, and already tested by
+      # `ProtobufIntegrationTest.test_import_from_buildroot`
       'testprojects/src/protobuf/org/pantsbuild/testproject/import_from_buildroot.*',
     ]
 
-    targets_to_exclude = (known_failing_targets + negative_test_targets + need_java_8 + python2_only +
-                          timeout_targets + deliberately_conflicting_targets + simply_skip)
-    exclude_opts = ['--exclude-target-regexp={}'.format(target) for target in targets_to_exclude]
+    return [
+      *known_failing_targets,
+      *negative_test_targets,
+      *need_java_8,
+      *python2_only,
+      *timeout_targets,
+      *deliberately_conflicting_targets,
+      *simply_skip,
+    ]
 
-    # Run list with exclude options, then parse and sort output.
-    pants_run = self.run_pants(['list', 'testprojects::', 'examples::'] + exclude_opts)
-    self.assert_success(pants_run)
-    return sorted(pants_run.stdout_data.split())
+  @memoized_property
+  def targets(self) -> List[str]:
+    """A sequence of target name strings."""
+
+    exclude_opts = [f'--exclude-target-regexp={target}' for target in self.skipped_targets]
+
+    def get_targets_for_type(build_file_alias: str) -> Set[str]:
+      pants_run = self.run_pants([
+        'filter',
+        f'--type={build_file_alias}',
+        'testprojects::',
+        'examples::',
+        *exclude_opts
+      ])
+      self.assert_success(pants_run)
+      return set(pants_run.stdout_data.split())
+
+    targets = set()
+    # TODO: map these build_file_aliases to more targeted goals than `test`. Right now, we use
+    #  `test` to trigger side effects via the goal graph, such as compiling JVM code or generating
+    #  antlr code. Instead, we should run the specific command we are testing for that target type.
+    #  For example, `python_antlr_library` would only need `./pants gen`.
+    for bfa in [
```

Comment:
Hm.

So, as discussed in Slack, I think that it would be better to do something like `./pants compile test lint` here (or whichever other goals) and wait for #8368 to land.

Reply:
Hm okay. I redesigned this PR around this feedback. Abandonded the goal of mapping target types -> commands; instead, everything now uses `./pants compile lint test`.

Also now blacklist certain target types, rather than this whitelist approach. IMO there is still value in skipping `python_requirement_library` and `files()` etc.